include/xgboost/json.h

```diff
@@ -223,6 +222,14 @@ class JsonObject : public Value {
   ~JsonObject() override = default;
 };
 
+namespace detail {
+template <typename T, typename U>
+using IsSameT = std::enable_if_t<std::is_same_v<std::remove_cv_t<T>, std::remove_cv_t<U>>>;
```

Comment:
Only extracting this due to a potential bug in clang-tidy-19, can't be sure yet. It keeps asking me to use modernized type traits if I don't add these helpers.